contracts/examples/adder/interact/src/basic_interact.rs

```diff
@@ -242,4 +250,4 @@ async fn test() {
     basic_interact.add(1u32).await;
 
     basic_interact.upgrade(7u32).await;
```

Comment:
Let's add a parameter of type ExpectError and have 2 transactions, one that fails as seen here (from non owner address) and one that's successful (from owner address) so that the test passes if run. Also upgrade and deploy need more than 3 mil gas in this case

Reply:
Upgraded the code a little, should work as intended now